Update dskit, add `-server.http-read-header-timeout` option (#6517)
grafana/mimir: CHANGELOG.md

* [ENHANCEMENT] Memcached: introduce new experimental configuration parameters `-<prefix>.memcached.write-buffer-size-bytes` `-<prefix>.memcached.read-buffer-size-bytes` to customise the memcached client write and read buffer size (the buffer is allocated for each memcached connection). #6468
* [ENHANCEMENT] Ingester, Distributor: added experimental support for rejecting push requests received via gRPC before reading them into memory, if ingester or distributor is unable to accept the request. This is activated by using `-ingester.limit-inflight-requests-using-grpc-method-limiter` for ingester, and `-distributor.limit-inflight-requests-using-grpc-method-limiter` for distributor. #5976 #6300
* [ENHANCEMENT] Query-frontend: return warnings generated during query evaluation. #6391

> * [ENHANCEMENT] Server: Add `-server.http-read-header-timeout` option to specify timeout for reading HTTP request header. It defaults to 0, in which case reading of headers can take up to `-server.http-read-timeout`, leaving no time for reading body, if there's any. #6517

Reviewer: ```suggestion
* [ENHANCEMENT] Server: Add the option `-server.http-read-header-timeout` to enable specifying a timeout for reading HTTP request headers. It defaults to 0, in which case reading headers can take up to the remaining time within `-server.http-read-timeout`. #6517
```

Author: > in which case reading headers can take up to the remaining time within `-server.http-read-timeout`.

It's not remaining, since HTTP request headers are read first. Remaining time is used to read body, if body needs to be read.

Author: I've applied first part of your suggestion, I'm not sure about the second sentence, so I left it as-is for now.
